### PR TITLE
Add docs for contributing

### PR DESCRIPTION
**Pull request by epicfaace**
### Reasons for making this change

Add docs for contributing on a separate page, and also explain the development / deployment process for the new multi-package setup.

### Checklist

* [x] **I'm updating documentation**
  - [x] I've [checked the rendering](https://react-jsonschema-form.readthedocs.io/en/latest/#contributing) of the Markdown text I've added
* [ ] **I'm adding or updating code**
  - [ ] I've added and/or updated tests
  - [ ] I've updated [docs](https://react-jsonschema-form.readthedocs.io/) if needed
* [ ] **I'm adding a new feature**
  - [ ] I've updated the playground with an example use of the feature

### CODE REVIEW

**Comment by epicfaace** [comment]
@juliankigwana @Saulzi @chanceaclark let me know if this is clearer or if the docs I've added might need more clarification?

**Comment by Saulzi** [comment]
Yes definitely clarifies things, I will give this workflow a go now

**Comment by Saulzi** [comment]
unfortunately this did not work on #1610

File: docs/contributing.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+Note that when doing local development in a package such as `packages/core`, making changes to `packages/playground` will not affect the local playground that is run from `packages/core`. This should be fine in most cases for development, unless you need to update and test the default playground examples. In this case, you could use `npm link` to use the latest changes both from the playground and your package. As an example, to do this when developing locally on `packages/core`, you can run:
+
+```bash
+cd packages/playground
+npm run build:lib
+npm link
+cd ../core
+npm link @rjsf/playground
+npm start
+```
```
Comment (Saulzi) [inline docs/contributing.md]: Perhaps this should be done when the user pulls the package rather than an additional step otherwise this works for me 👍

**Comment by Saulzi** [comment]
> unfortunately this did not work on #1610

this did after install on bottom level and setting back index.js to be import Form, { withTheme } from "@rjsf/core";  in packages/playground on my branch

**Comment by epicfaace** [comment]
Would it be simpler, rather than having each package have its own playground start scripts, to just have each package build their files (as @chanceaclark 's PR in #1606 does)? Then, in order to run the playground, one would just need to `npm link` whichever package they are working on, then run the playground from the `packages/playground` directory.

This thought came up because I was trying to add the playground to material-ui but realized that doing so would require a lot of overhead for each new package we add.

**Pull request merged**